core/metacling/src/TCling.cxx

```diff
@@ -351,6 +351,29 @@ void TCling__PrintStackTrace() {
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// Lock the interpreter.
+
+extern "C"
+void TCling__RestoreInterpreterMutex(void* state) {
+   if (gInterpreterMutex && state) {
+      std::unique_ptr<TVirtualMutex::State> uniqueP{static_cast<TVirtualMutex::State*>(state)};
```

Comment:
Would it be more readable with
```
auto uniqueP{static_cast<TVirtualMutex::State*>(state)};
```

Reply:
More readable yes, but sadly also ill-formed: I need a `unique_ptr`.

Reply:
which proves it is hard to read .. I obviously did not really noticed the missing unique_ptr in the cast :( 
What about
```c++
void TCling__RestoreInterpreterMutex(void *state) {
   if (gInterpreterMutex && state) {
      auto typedState = static_cast<TVirtualMutex::State*>(state);
      std::unique_ptr<TVirtualMutex::State> uniqueP{typeState};
      gInterpreterMutex->Restore(std::move(uniqueP));
   }
}
```

and you shouldn't use that but the following ought to work also

```
void TCling__RestoreInterpreterMutex(void* state) {
   if (gInterpreterMutex && state) {
      // unique_ptr is guaranteed to be the same size and content as void*
      auto uniquePtrPtr = reinterpret_cast<std::unique_ptr<TVirtualMutex::State> >( &state );
      gInterpreterMutex->Restore(std::move(*uniquePtrPtr));
   }
}
```